### PR TITLE
Switched over to using data from the database for swimming units

### PR DESCRIPTION
**Pull request by orangemug**
## Description
Switched over to using `{"pe_swimming": true}` data from the database for swimming units

**Note**: Waiting on merge/release of https://github.com/oaknational/Database-Tools/pull/436

## Issue(s)

Fixes `CUR-842`

## How to test
Enable cycle 2 and check correct swimming units are displayed.

### CODE REVIEW

File: src/utils/curriculum/features.ts [hunk 1]
```diff
@@ -32,14 +24,7 @@ export function getUnitFeatures(unit?: Unit | null) {
...
 
-  // HACK: Swimming primary isn't yet published so we're hacking in some secondary units and giving them the overrides
-  if (
-    isSwimmingHackEnabled() &&
-    [
-      "health-and-wellbeing-hiit-and-couch-to-5k-team-challenges-to-develop-fitness",
-      "sport-psychology-skill-and-ability",
-    ].includes(unit.slug)
-  ) {
+  if (unit.features?.pe_swimming) {
```
Comment (rightsaidjames) [inline src/utils/curriculum/features.ts:27]: Excuse my ignorance, but does this check that `pe_swimming` is true, or that the key exists?

File: src/utils/curriculum/features.test.ts [hunk 2]
```diff
@@ -87,25 +65,8 @@ describe("isCurricPartnerHackEnabled", () => {
 });
 
 describe("getUnitFeatures", () => {
-  it("returns swimming rules when hack enabled and matching unit", () => {
```
Comment (rightsaidjames) [inline src/utils/curriculum/features.test.ts:90]: Do we need a replacement test for the new behaviour?

File: src/utils/curriculum/features.ts (hunk 1 above)
Comment (rightsaidjames) [reply src/utils/curriculum/features.ts:27]: (I looked this up myself, all good!)

**Review by sonalisin — APPROVED** [review]

File: src/utils/curriculum/features.test.ts (hunk 2 above)
Comment (orangemug) [reply src/utils/curriculum/features.test.ts:90]: I've added this back in now

**Review by rightsaidjames — APPROVED** [review]

**Pull request merged**